iac-extensions/terraform/src/test/resources/checks/Azure/CertificateBasedAuthenticationCheck/function_app.tf

```diff
@@ -0,0 +1,15 @@
+resource "azurerm_function_app" "sensitive1" {
+  client_cert_mode = "Optional" # Noncompliant {{Make sure that disabling certificate-based authentication is safe here.}}
```

Comment:
Normally, I would expect to see some control comments which test the highlighting. However, you used the reportOn methods of AbstractResourceCheck and they only highlight the attribute value and not the entire attribute. So your current highlighting is not compliant with the specification. I've created SONARIAC-286 to fix this and I will also fix your check during this ticket.